itachallenge-challenge/src/main/java/com/itachallenge/challenge/mapper/submission/SubmissionResponseDtoMapper.java

```
@@ -0,0 +1,25 @@
+package com.itachallenge.challenge.mapper.submission;
+
+
+import com.itachallenge.challenge.dto.submission.SubmissionResponseDto;
+import com.itachallenge.submission.document.SubmissionDocument;
+
+public final class SubmissionResponseDtoMapper {
+
+    private SubmissionResponseDtoMapper() {
+    }
+
+    public static SubmissionResponseDto toDto(SubmissionDocument doc) {
+        if (doc == null) {
+            return null;
```

Returning null is not a good idea. See other options.

---

You're right, the DTO is pointless without the document; we opted to fail fast.